src/plugins/visualize/public/plugin.ts

```diff
@@ -112,39 +112,51 @@ export class VisualizePlugin
         const [coreStart, pluginsStart] = await core.getStartServices();
         this.currentHistory = params.history;
 
+        // make sure the index pattern list is up to date
+        pluginsStart.data.indexPatterns.clearCache();
```

Comment:
@sulemanof  could you please explain me for which case this code was added? I just want to realize why we need to clear internal for data plugin cache. 

Interesting, but the same code we also have in `Dashboard / Discovery` plugins in start hook. But from my point of view it's doesn't have a lot of sense. 

@lizozom  What do you think about it?

Reply:
I'm not really sure why this was added (not in my PR), I could only operate with the comment left 😄 
I assume @flash1293 can answer this question

Reply:
This fixed several issues, when an index pattern was deleted, but the cache wasn't refreshed.
Here's the PR for that #63329